doc/radosgw/multisite.rst

```diff
@@ -0,0 +1,710 @@
+objects etc. can be handled by any of the zones.
+
+System Keys
+===========
+
+While configuring zones, RadosGW, expects creation of zone a system user, with
+an S3 like access and secret keys. This allows another radosgw instance to pull
+the configuration remotely, with the access and secret keys. For making
+scripting and use of configuration management tools easier, it would be easier
+to generate the access-key and secret-key before hand and use these. For the
+purpose of the guide we assume an access key and secret key are set in the
+environment. For eg., something like::
+
+  $ SYSTEM_ACCESS_KEY=1555b35654ad1656d805
+  $ SYSTEM_SECRET_KEY=h7GhxuBLTrlhVUyxSPUKUV8r/2EI4ngqJxD7iBdBYLhwluN30JaT3Q==
+
+Naming conventions
+==================
+
+This section describes the process of setting up a master zone. For the purpose
+of the guide, we'll assume a zonegroup called ``us`` spanning the United States,
+which will be our master zonegroup. This will contain two zones written in a
+``{zonegroup}-{zone}`` format, this is just a convention only and you are free
+to choose a format you prefer to set these names. So in summary:
+
+- Master Zonegroup: United States ``us``
+- Master zone: United States, East Region 1: ``us-east-1``
+- Secondary zone: United States, East Region 2: ``us-east-2``
+- Secondary zone: United States, West Region: ``us-west``
+
+This will be a part of a larger realm, say ``gold``. The zones ``us-east-1`` and
+``us-east-2`` are part of the same ceph cluster, of which we'll make
+``us-east-1`` as the primary. ``us-west`` will be in a different ceph cluster.
+
+Creating a realm
+----------------
+
+Configure a realm called ``gold``, and also make this the default ::
+
+  # radosgw-admin realm create --rgw-realm=gold --default
+  {
+    "id": "4a367026-bd8f-40ee-b486-8212482ddcd7",
+    "name": "gold",
+    "current_period": "09559832-67a4-4101-8b3f-10dfcd6b2707",
+    "epoch": 1
+  }
+
+
+Note that every realm has an id, which allows for flexibility like renaming the
+realm later, should the need arise. The ``current_period`` changes whenever we
+change anything in the master zone. The ``epoch`` is incremented when there's a
+change in configuration of the master zone, which results in a change of current
+period.
+
+Deleting the default zonegroup
+------------------------------
+
+A simple installation of radosgw would assume the default zonegroup called
+"default", since we no longer need this we begin by removing the default
+zonegroup::
+
+  # radosgw-admin zonegroup delete --rgw-zonegroup=default
+
+
+Creating a master zonegroup
+---------------------------
+
+We'll be creating a zonegroup called ``us`` as a master zonegroup. A master
+zonegroup will be in control of the zonegroup map and propagate changes to the
+rest of the system. We will also set this zonegroup as the default, which allows
+explicitly mentioning the ``rgw-zonegroup`` switch for later commands. We also
+make this the default zonegroup
```

Comment:
> We will also set this zonegroup as the default, which allows explicitly mentioning the ``rgw-zonegroup`` switch for later commands. We also make this the default zonegroup

We will also set this zonegroup as the default, which allows later commands to use this zonegroup without explicitly mentioning it with the ``rgw-zonegroup`` switch.